store/src/state.rs

```diff
@@ -479,10 +479,18 @@ impl State {
 // ================================================================================================
 
 /// Returns the nullifier's leaf value in the SMT by its block number.
-fn block_to_nullifier_data(block: BlockNumber) -> Word {
+fn block_num_to_nullifier_value(block: BlockNumber) -> Word {
     [Felt::from(block), Felt::ZERO, Felt::ZERO, Felt::ZERO]
 }
 
+/// Given the leaf value of the nullifier SMT, returns the nullifier's block number.
+///
+/// There are no nullifiers in the genesis block. The value zero is instead used to signal absence
+/// of a value.
+pub fn nullifier_value_to_block_num(value: Word) -> BlockNumber {
+    value[0].as_int().try_into().expect("invalid block number found in store")
```

Comment:
Oh, I see, the bug was using `3` instead of `0` here?

Lets please add a test for this.

Reply:
@hackaugusto I've added tests.